src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1196,14 +1196,16 @@ internal NamedTypeSymbol GetWellKnownType(WellKnownType type, DiagnosticBag diag
 
         /// <summary>
         /// Retrieves a well-known type member and reports diagnostics.
+        ///
+        /// If a type is passed in, then the well-known member will be looked up within that type.
         /// </summary>
         /// <returns>Null if the symbol is missing.</returns>
-        internal static Symbol GetWellKnownTypeMember(CSharpCompilation compilation, WellKnownMember member, DiagnosticBag diagnostics, Location location = null, CSharpSyntaxNode syntax = null, bool isOptional = false)
+        internal static Symbol GetWellKnownTypeMember(CSharpCompilation compilation, WellKnownMember member, DiagnosticBag diagnostics, Location location = null, CSharpSyntaxNode syntax = null, bool isOptional = false, NamedTypeSymbol type = null)
```

Comment:
Ok, I got confused about what helper is this. I think we shouldn't modify it at all because it is very uncommon to lookup members like that. Basically, the new lookup is specific to tuples and we should keep it this way. Besides the diagnostics reported by this method is completely based off of descriptor, including the information about containing type. I think for the tuple case we would want to report a different error.